dev/com.ibm.ws.concurrent.mp_fat/test-applications/MPConcurrentTxApp/src/concurrent/mp/fat/tx/web/MPConcurrentTxTestServlet.java

```diff
@@ -1046,6 +1046,130 @@ public void testResourceLocalTransactionWithUnsharedHandleUsedSeriallyAndRollBac
         }
     }
 
+    /**
+     * Propagates transaction context to be used serially from another thread, which also commits the transaction.
+     * The resource that is enlisted in the transaction is a sharable one-phase only resource.
+     * A single connection handle is cached and reused across all stages where transactional work is performed.
+     */
+    @Test
+    public void testSharableOnePhaseResourceInJTATransactionCachedHandleUsedSeriallyAndCommitWhenComplete() throws Exception {
+        CompletableFuture<DataSource> getDataSource = txExecutor.newIncompleteFuture();
+        CompletableFuture<Integer> stage;
+        tx.begin();
+        try {
+            stage = getDataSource.thenApplyAsync(ds -> {
+                try {
+                    Connection con = ds.getConnection();
+                    con.createStatement().executeUpdate("INSERT INTO MNCOUNTIES VALUES ('Mille Lacs', 25635)");
+                    return con;
+                } catch (SQLException x) {
+                    throw new CompletionException(x);
+                }
+            }).thenApplyAsync(con -> {
+                try {
+                    con.createStatement().executeUpdate("INSERT INTO MNCOUNTIES VALUES ('Morrison', 32880)");
+                    return con;
+                } catch (SQLException x) {
+                    throw new CompletionException(x);
+                }
+            }).thenApplyAsync(con -> {
+                try {
+                    return con.createStatement().executeUpdate("INSERT INTO MNCOUNTIES VALUES ('Murray', 8394)");
```

Comment:
Should we be closing this connection explicitly?

Reply:
No, that is one of the advantages of using sharable connections: you shouldn't need to close handles within a transaction because at the end of the transaction boundary they will all be dissociated.  This test case is intentionally relying upon this, and we should be able to look at the connection pool after running the test and confirm that no connections are still allocated.